fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/models/FedoraResourceImpl.java

```diff
@@ -250,6 +251,9 @@ public boolean hasType(final String type) {
             // ldp:Resource is on all resources
             types.add(RESOURCE_URI);
             types.add(FEDORA_RESOURCE_URI);
+            if (getFedoraId().isRepositoryRoot()) {
+                types.add(create(REPOSITORY_ROOT.getURI()));
```

Comment:
Rather than creating the URI every time, just create it once as a static constant